app/DynamicGraphs/GraphGenerator.hs

```diff
@@ -64,11 +65,38 @@ sampleGraph = fst $ State.runState (reqsToGraph
 --  multiple Reqs using the same Grade requirement
 reqsToGraph :: GraphOptions -> [(Text, Req)] -> State GeneratorState (DotGraph Text)
 reqsToGraph options reqs = do
-    allStmts <- concatUnique <$> mapM (reqToStmts options) reqs
+    allStmts <- concatUnique <$> mapM (reqToStmts options) filteredReqs
     return $ buildGraph allStmts
     where
         concatUnique = nubOrd . Prelude.concat
-
+        filteredReqs = [(t, filteredReq) | (t, req) <- reqs,
+                                           let filteredReq = filterReq options req,
+                                           filteredReq /=None]
+                                           
+
+-- | Recurse through the Req Tree to remove any nodes specified in GraphOptions
+filterReq :: GraphOptions -> Req -> Req
+filterReq options None = None
+filterReq options (J course info)
+    | not (Prelude.null (departments options)) && not (any (`isPrefixOf` pack course) (departments options)) = None
+    | pack course `elem` taken options = None
+    | otherwise = J course info
+filterReq options (ReqAnd reqs) =
+    case Prelude.filter (/= None) (map (filterReq options) reqs) of
+        [] -> None
+        [r] -> r
+        reqs' -> ReqAnd reqs'
+filterReq options (ReqOr reqs) =
+    case Prelude.filter (/= None) (map (filterReq options) reqs) of
+        [] -> None
+        [r] -> r
+        reqs' -> ReqOr reqs'
+filterReq options (Fces fl modifier) = Fces fl modifier
```

Comment:
for any pattern that doesn't use `options`, replace the parameter name with `_` so that you don't get a compiler warning